packages/belos/tpetra/example/PCPG/PCPGTpetraExFile.cpp

```diff
@@ -0,0 +1,437 @@
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+
+// Purpose
+// The example tests the successive right-hand sides capabilities of ML
+// and Belos on a heat flow u_t = u_xx problem.
+//
+// A sequence of linear systems with the same coefficient matrix and
+// different right-hand sides is solved.  A seed space is generated dynamically,
+// and a deflated linear system is solved.  After each solves, the first
+// few Krylov vectors are saved, and used to reduce the number of iterations
+// for later solves.
+// The optimal numbers of vectors to deflate and save are not known.
+// Presently, the maximum number of vectors to deflate (seed space dimension)
+// and to save are user paraemters.
+// The seed space dimension is less than or equal to total number of vectors
+// saved. The difference between the seed space dimension and the total number
+// of vectors, is the number of vectors used to update the seed space after each
+// solve. I guess that a seed space whose dimension is a small fraction of the
+// total space will be best.
+//
+// maxSave=1 and maxDeflate=0 uses no recycling (not tested ).
+//
+// TODO: Instrument with timers, so that we can tell what is going on besides
+//       by counting the numbers of iterations.
+//
+//
+// \author David M. Day
+//
+// \data Last modified 2007 December 11
+
+// Tpetra
+#include <TpetraExt_MatrixMatrix.hpp>
+#include <Tpetra_Core.hpp>
+#include <Tpetra_CrsMatrix_fwd.hpp>
+#include <Tpetra_Map_fwd.hpp>
+#include <Tpetra_Vector_fwd.hpp>
+
+// MueLu
+// #include <MueLu_TpetraOperator.hpp>
+// #include <MueLu_CreateTpetraPreconditioner.hpp>
+
+// Teuchos
+#include <Teuchos_Comm.hpp>
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_FancyOStream.hpp>
+#include <Teuchos_ParameterList.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
+#include <Teuchos_Tuple.hpp>
+#include <Teuchos_VerboseObject.hpp>
+
+// Belos
+#include "BelosLinearProblem.hpp"
+#include "BelosPCPGSolMgr.hpp"
+#include "BelosTpetraAdapter.hpp"
+
+template <typename ScalarType>
+int run(int argc, char *argv[]) {
+  //
+  // Laplace's equation, homogeneous Dirichlet boundary conditions, [0,1]^2
+  // regular mesh, Q1 finite elements
+  //
+
+  using std::cout;
+  using std::endl;
```

Comment:
It's often a bad practice to do this: the prefixes objectively add clarity to the code. See more informations here: https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice

Reply:
In fact the prefixes were used and these using are not useful. Removed.